src/libraries/Microsoft.Extensions.Caching.Memory/src/CacheEntry.cs

```diff
@@ -28,7 +28,7 @@ internal CacheEntry(object key, MemoryCache memoryCache)
         {
             Key = key ?? throw new ArgumentNullException(nameof(key));
             _cache = memoryCache ?? throw new ArgumentNullException(nameof(memoryCache));
-            _previous = CacheEntryHelper.EnterScope(this);
+            _previous = memoryCache.TrackLinkedCacheEntries ? CacheEntryHelper.EnterScope(this) : null;
```

Comment:
Need to update the comment above on `_previous`'s declaration:

> private CacheEntry _previous; // this field is not null only before the entry is added to the cache

That comment is now wrong.

Reply:
good catch